miner/worker.go

```diff
@@ -1196,8 +1196,42 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 		if w.commitBundle(bundleTxs, w.coinbase, interrupt) {
 			return
 		}
-		w.current.profit.Add(w.current.profit, bundle.totalEth)
+		w.current.profit.Add(w.current.profit, bundle.ethSentToCoinbase)
 	}
+	if w.flashbots.isMegabundleWorker {
+		megabundle, err := w.eth.TxPool().GetMegabundle(w.flashbots.relayAddr, header.Number, header.Time)
+		log.Info("Starting to process a Megabundle", "relay", w.flashbots.relayAddr, "megabundle", megabundle, "error", err)
+		if err != nil {
+			return // no valid megabundle for this relay, nothing to do
+		}
+		// Flashbots bundle merging duplicates work by simulating TXes and then committing them once more.
+		// Megabundles API focuses on speed and runs everything in one cycle.
+		coinbaseBalanceBefore := w.current.state.GetBalance(w.coinbase)
+		if w.commitBundle(megabundle.Txs, w.coinbase, interrupt) {
+			log.Info("Could not commit a Megabundle", "relay", w.flashbots.relayAddr, "megabundle", megabundle)
+			return
+		}
+		var txStatuses = map[common.Hash]bool{}
+		for _, receipt := range w.current.receipts {
+			txStatuses[receipt.TxHash] = receipt.Status == types.ReceiptStatusSuccessful
+		}
+		for _, tx := range megabundle.Txs {
+			status, ok := txStatuses[tx.Hash()]
+			if !ok {
+				log.Error("No TX receipt after megabundle simulation", "TxHash", tx.Hash())
+				return
+			}
+			if !status && !containsHash(megabundle.RevertingTxHashes, tx.Hash()) {
+				log.Info("Ignoring megabundle because of failing TX", "relay", w.flashbots.relayAddr, "TxHash", tx.Hash())
+				return
+			}
+		}
+		coinbaseBalanceAfter := w.current.state.GetBalance(w.coinbase)
+		coinbaseDelta := big.NewInt(0).Sub(coinbaseBalanceAfter, coinbaseBalanceBefore)
+		w.current.profit = coinbaseDelta
```

Comment:
Would this profit calculation be off if the miner sends (or receives) transactions with value from/to the coinbase address?